### ISSUE TITLE
CentOS7 installation error

### ISSUE THREAD

**Issue by 1duo** [horovod/horovod#386]
```
$ pip install horovod
Collecting horovod
  Using cached https://files.pythonhosted.org/packages/2c/7e/54f7759bc693ab19480c1b525decab5c1d846cd5fc3af97bda18364a1aa5/horovod-0.13.10.tar.gz
Requirement already satisfied: cffi>=1.4.0 in /usr/lib64/python2.7/site-packages (from horovod) (1.6.0)
Requirement already satisfied: pycparser in /usr/lib/python2.7/site-packages (from cffi>=1.4.0->horovod) (2.14)
Building wheels for collected packages: horovod
  Running setup.py bdist_wheel for horovod ... error
  Complete output from command /usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-MkrWxc/horovod/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-hqPEwf --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-2.7
  creating build/lib.linux-x86_64-2.7/horovod
  copying horovod/__init__.py -> build/lib.linux-x86_64-2.7/horovod
  creating build/lib.linux-x86_64-2.7/horovod/keras
  copying horovod/keras/__init__.py -> build/lib.linux-x86_64-2.7/horovod/keras
  copying horovod/keras/callbacks.py -> build/lib.linux-x86_64-2.7/horovod/keras
  creating build/lib.linux-x86_64-2.7/horovod/torch
  copying horovod/torch/mpi_ops.py -> build/lib.linux-x86_64-2.7/horovod/torch
  copying horovod/torch/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch
  creating build/lib.linux-x86_64-2.7/horovod/common
  copying horovod/common/__init__.py -> build/lib.linux-x86_64-2.7/horovod/common
  creating build/lib.linux-x86_64-2.7/horovod/tensorflow
  copying horovod/tensorflow/mpi_ops.py -> build/lib.linux-x86_64-2.7/horovod/tensorflow
  copying horovod/tensorflow/__init__.py -> build/lib.linux-x86_64-2.7/horovod/tensorflow
  creating build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib
  copying horovod/torch/mpi_lib/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib
  creating build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib_impl
  copying horovod/torch/mpi_lib_impl/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib_impl
  running build_ext
  gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -std=c++11 -fPIC -O2 -I/usr/include/python2.7 -c build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.cc -o build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.o
  gcc -pthread -shared -Wl,-z,relro build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.o -L/usr/lib64 -o build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.so
  error: mpicxx -show failed (see error below), is MPI in $PATH?
  Note: If your version of MPI has a custom command to show compilation flags, please specify it with the HOROVOD_MPICXX_SHOW environment variable.
  
  Traceback (most recent call last):
    File "/tmp/pip-install-MkrWxc/horovod/setup.py", line 193, in get_mpi_flags
      shlex.split(show_command), universal_newlines=True).strip()
    File "/usr/lib64/python2.7/subprocess.py", line 568, in check_output
      process = Popen(stdout=PIPE, *popenargs, **kwargs)
    File "/usr/lib64/python2.7/subprocess.py", line 711, in __init__
      errread, errwrite)
    File "/usr/lib64/python2.7/subprocess.py", line 1327, in _execute_child
      raise child_exception
  OSError: [Errno 2] No such file or directory
  
  
  ----------------------------------------
  Failed building wheel for horovod
  Running setup.py clean for horovod
Failed to build horovod
ipaclient 4.5.4 requires jinja2, which is not installed.
rtslib-fb 2.1.63 has requirement pyudev>=0.16.1, but you'll have pyudev 0.15 which is incompatible.
ipapython 4.5.4 has requirement dnspython>=1.15, but you'll have dnspython 1.12.0 which is incompatible.
Installing collected packages: horovod
  Running setup.py install for horovod ... error
    Complete output from command /usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-MkrWxc/horovod/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ZjzybR/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    creating build/lib.linux-x86_64-2.7/horovod
    copying horovod/__init__.py -> build/lib.linux-x86_64-2.7/horovod
    creating build/lib.linux-x86_64-2.7/horovod/keras
    copying horovod/keras/__init__.py -> build/lib.linux-x86_64-2.7/horovod/keras
    copying horovod/keras/callbacks.py -> build/lib.linux-x86_64-2.7/horovod/keras
    creating build/lib.linux-x86_64-2.7/horovod/torch
    copying horovod/torch/mpi_ops.py -> build/lib.linux-x86_64-2.7/horovod/torch
    copying horovod/torch/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch
    creating build/lib.linux-x86_64-2.7/horovod/common
    copying horovod/common/__init__.py -> build/lib.linux-x86_64-2.7/horovod/common
    creating build/lib.linux-x86_64-2.7/horovod/tensorflow
    copying horovod/tensorflow/mpi_ops.py -> build/lib.linux-x86_64-2.7/horovod/tensorflow
    copying horovod/tensorflow/__init__.py -> build/lib.linux-x86_64-2.7/horovod/tensorflow
    creating build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib
    copying horovod/torch/mpi_lib/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib
    creating build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib_impl
    copying horovod/torch/mpi_lib_impl/__init__.py -> build/lib.linux-x86_64-2.7/horovod/torch/mpi_lib_impl
    running build_ext
    gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -std=c++11 -fPIC -O2 -I/usr/include/python2.7 -c build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.cc -o build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.o
    gcc -pthread -shared -Wl,-z,relro build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.o -L/usr/lib64 -o build/temp.linux-x86_64-2.7/test_compile/test_cpp_flags.so
    error: mpicxx -show failed (see error below), is MPI in $PATH?
    Note: If your version of MPI has a custom command to show compilation flags, please specify it with the HOROVOD_MPICXX_SHOW environment variable.
    
    Traceback (most recent call last):
      File "/tmp/pip-install-MkrWxc/horovod/setup.py", line 193, in get_mpi_flags
        shlex.split(show_command), universal_newlines=True).strip()
      File "/usr/lib64/python2.7/subprocess.py", line 568, in check_output
        process = Popen(stdout=PIPE, *popenargs, **kwargs)
      File "/usr/lib64/python2.7/subprocess.py", line 711, in __init__
        errread, errwrite)
      File "/usr/lib64/python2.7/subprocess.py", line 1327, in _execute_child
        raise child_exception
    OSError: [Errno 2] No such file or directory
    
    
    ----------------------------------------
Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-MkrWxc/horovod/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ZjzybR/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-MkrWxc/horovod/
```

Error says: `error: mpicxx -show failed (see error below), is MPI in $PATH?`, but I have OpenMPI 3.1.0 installed:

```
$ mpirun --version
mpirun (Open MPI) 3.1.0

Report bugs to http://www.open-mpi.org/community/help/
```

and path already set in the `$PATH`:

```
$ which mpicc
/usr/local/bin/mpicc
$ echo $PATH
/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/novuforce/.local/bin:/home/novuforce/bin:/usr/local/bin/
```

Any idea what is the root cause of this installation error?

**Comment by 1duo** [comment]
```
$ mpicxx -show
/usr/local/bin/g++ -I/usr/local/include -pthread -Wl,-rpath -Wl,/usr/local/lib -Wl,--enable-new-dtags -L/usr/local/lib -lmpi
```

**Comment by alsrgv** [comment]
@1duo, that's very strange.  I just installed tensorflow + Open MPI 3.1.0 + horovod in CentOS 7 docker without issues.

Are you using any virtual environments in your installation?

Did you try reopening your bash session after installing Open MPI?  Sometimes bash needs to re-index $PATH.

**Comment by 1duo** [comment]
Hi @alsrgv , which docker base image are you using?

**Comment by alsrgv** [comment]
@1duo, `centos:7`

**Comment by 1duo** [comment]
@alsrgv I tried with docker base image from `nvidia/cuda:9.1-cudnn7-devel-centos7`, seems works fine: https://gist.github.com/1duo/43acb5349578a414291195f892ed874b. Still don't know why on my host machine it fails. I do not have any virtual environment in that case.

**Comment by mohammedayub44** [comment]
I got the same error installing in the User directory I followed this blog post : 
`https://superuser.com/questions/927512/how-to-set-path-for-sudo-commands`
to edit the sudoers $PATH because pip overrides user's $PATH variable.
and then ran 
 `sudo pip install --verbose --no-cache-dir horovod`

and it worked fine. 
I'm not sure this is the most secure way to do it, as not everyone will have access to edit the sudoers $PATH file.
